MH4U Dex/CoreDataController.m

```diff
@@ -46,88 +48,79 @@ + (CoreDataController *)sharedCDController
-                   DamageZoneAttributes.jpnMonsterName: @"jpn_monster_name",
-                   DamageZoneAttributes.bodyPart: @"body_part",
-                   DamageZoneAttributes.cut: @"cut",
-                   DamageZoneAttributes.impact: @"impact",
-                   DamageZoneAttributes.shot: @"shot",
-                   DamageZoneAttributes.fire: @"fire",
-                   DamageZoneAttributes.water: @"water",
-                   DamageZoneAttributes.thunder: @"thunder",
-                   DamageZoneAttributes.ice: @"ice",
-                   DamageZoneAttributes.dragon: @"dragon",
-                   DamageZoneAttributes.ko: @"ko",
-                   DamageZoneAttributes.extract: @"extract",
-                   };
-    [self.managedObjectContext hydrateStoreWithJSONAtPath:path attributeMappings:attributes forEntityName:[DamageZone entityName]];
+    NSArray *monsterList = [self loadArrayFromJsonFileNamed:MHDMonstersFileName];
+    for (NSDictionary *monsterDict in monsterList) {
+        Monster *monster = [self monsterWithName:monsterDict[MHDMonsterNameKey]];
+        if (!monster) {
+            monster = [Monster insertInManagedObjectContext:self.managedObjectContext];
+        }
+        monster.name = monsterDict[MHDMonsterNameKey];
+        monster.monster_description = monsterDict[MHDMonsterDescriptionKey];
+        monster.sort_name = monsterDict[MHDMonsterSortNameKey];
+        monster.monster_class = monsterDict[MHDMonsterClassKey];
+    }
 }
 
 - (void)loadMonsterDropData
 {
-    NSArray *dropList = [self loadArrayFromJsonFileNamed:MHDMonsterDropsFileName];
-    for (NSDictionary *drop in dropList) {
-        // Only add the drop if it is not already there
-        NSNumber *monsterDropID = drop[MHDMonsterDropIDKey];
-        NSPredicate *monsterDropPredicate = [NSPredicate predicateWithFormat:@"%K == %d", MonsterAttributes.id, monsterDropID.intValue];
-        if (![self countForEntityWithEntityName:[MonsterDrop entityName] withPredicate:monsterDropPredicate]) {
-            MonsterDrop *newDrop = [MonsterDrop insertInManagedObjectContext:self.managedObjectContext];
-            newDrop.id = drop[MHDMonsterDropIDKey];
-            newDrop.method = drop[MHDMonsterDropMethodKey];
-            newDrop.quantity = drop[MHDMonsterDropQuantityKey];
-            
-            // If the Item for this drop already exists in the persistent store ...
-            NSPredicate *itemPredicate = [NSPredicate predicateWithFormat:@"%K == %@", ItemAttributes.name, drop[MHDMonsterDropItemNameKey]];
-            Item *item = (Item *)[self uniqueEntityWithEntityName:[Item entityName] withPredicate:itemPredicate];
-            if (!item) {
-                item = [Item insertInManagedObjectContext:self.managedObjectContext];
-                item.name = drop[MHDMonsterDropItemNameKey];
+    //Repeat this process for ten different JSON files
+    for (NSUInteger fileNumber = 1; fileNumber <= 10; fileNumber++) {
```

Comment:
Even if you're using static consts, those at least are more expressive - they tell you *what* 1 and 10 are rather than just that the numbers are 1 and 10.